app/modules/xpt2046.c

```diff
@@ -0,0 +1,213 @@
+  else reta = (y + z) >> 1;   
+
+  return reta;
+}
+
+// Checks if the irq_pin is down
+static int ICACHE_FLASH_ATTR isTouching() { 
+	return (platform_gpio_read(_irq_pin) == 0); 
+}
+
+// transfer 16 bits from the touch display - returns the recived uint16_t
+static uint16_t ICACHE_FLASH_ATTR transfer16(uint16_t _data) {
+	union { uint16_t val; struct { uint8_t lsb; uint8_t msb; }; } t;
+	t.val = _data;
+	t.msb = platform_spi_send_recv(1, 8 , t.msb);
+	t.lsb = platform_spi_send_recv(1, 8 , t.lsb);
+	return t.val;
+}
+
+// reads the value from the touch panel
+static uint16_t ICACHE_FLASH_ATTR _readLoop(uint8_t ctrl, uint8_t max_samples) {
+  uint16_t prev = 0xffff, cur = 0xffff;
+  uint8_t i = 0;
+  do {
+    prev = cur;
+    cur = platform_spi_send_recv(1, 8 , 0);
+    cur = (cur << 4) | (platform_spi_send_recv(1, 8 , ctrl) >> 4);  // 16 clocks -> 12-bits (zero-padded at end)
+  } while ((prev != cur) && (++i < max_samples));
+  return cur;
+}
+
+// returns the raw position information
+static void ICACHE_FLASH_ATTR getRaw(uint16_t *vi, uint16_t *vj) {
+  // Implementation based on TI Technical Note http://www.ti.com/lit/an/sbaa036/sbaa036.pdf
+
+  platform_gpio_write(_cs_pin, 0);
+  platform_spi_send_recv(1, 8 , CTRL_HI_X | CTRL_LO_DFR);  // Send first control int
+  *vi = _readLoop(CTRL_HI_X | CTRL_LO_DFR, 255);
+  *vj = _readLoop(CTRL_HI_Y | CTRL_LO_DFR, 255);
+
+  // Turn off ADC by issuing one more read (throwaway)
+  // This needs to be done, because PD=0b11 (needed for MODE_DFR) will disable PENIRQ
+  platform_spi_send_recv(1, 8 , 0);  // Maintain 16-clocks/conversion; _readLoop always ends after issuing a control int
+  platform_spi_send_recv(1, 8 , CTRL_HI_Y | CTRL_LO_SER);
+  transfer16(0);  // Flush last read, just to be sure
+  
+  platform_gpio_write(_cs_pin, 1);
+}
+
+// sets the calibration of the display
+static void ICACHE_FLASH_ATTR setCalibration (uint16_t vi1, uint16_t vj1, uint16_t vi2, uint16_t vj2) {
+  _cal_dx = _width - 2*CAL_MARGIN;
+  _cal_dy = _height - 2*CAL_MARGIN;
+
+  _cal_vi1 = (int32_t)vi1;
+  _cal_vj1 = (int32_t)vj1;
+  _cal_dvi = (int32_t)vi2 - vi1;
+  _cal_dvj = (int32_t)vj2 - vj1;
+}
+
+// returns the position on the screen by also applying the calibration
+static void ICACHE_FLASH_ATTR getPosition (uint16_t *x, uint16_t *y) {
+  if (isTouching() == 0) {
+    *x = *y = 0xffff;
+    return;
+  }
+
+  uint16_t vi, vj;
+  getRaw(&vi, &vj);
+
+  // Map to (un-rotated) display coordinates
+  *x = (uint16_t)(_cal_dx * (vj - _cal_vj1) / _cal_dvj + CAL_MARGIN);
+  *y = (uint16_t)(_cal_dy * (vi - _cal_vi1) / _cal_dvi + CAL_MARGIN);
+}
+
+
+// Lua: xpt2046.setup(cspin,irqpin,height,width)
+static int ICACHE_FLASH_ATTR xpt2046_setup( lua_State* L ) {
+  _cs_pin  = luaL_checkinteger( L, 1 );
+  _irq_pin = luaL_checkinteger( L, 2 );
+  _height  = luaL_checkinteger( L, 3 );
+  _width  = luaL_checkinteger( L, 4 );
+  // set pins correct
+  platform_gpio_mode(_cs_pin  , 1, 0);; //output
```

Comment:
It's advisable to use the macros from `platform.h` like `PLATFORM_GPIO_OUTPUT` and `PLATFORM_GPIO_HIGH` instead of literals.